### PR TITLE
Feature/asp net

### PR DESCRIPTION
**Pull request by ademar**

### CODE REVIEW

File: src/Suave.Xsp/Library.fs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+open Suave
+open Suave.Types
+open Suave.Types.Codes
+open Suave.Http
+open Suave.Utils
+
+open Suave.Sockets
+
+module Xsp =
```
Comment (dsyme) [inline src/Suave.Xsp/Library.fs]: Could you add a short summary comment on each module and type definition, and preferably each member too?  For example what Xsp stands for  (I know it's going to be eXtreme Server Pages :) ) thanks

File: src/Suave.Xsp/Library.fs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  type Connection(transport : ITransport) =
+    inherit MarshalByRefObject()
```
Comment (dsyme) [inline src/Suave.Xsp/Library.fs]: Could you add a comment about why this needs to be MarshalByRef?  I don't see that used so much anymore so am curious.

File: src/Suave.Xsp/Script.fsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿// Learn more about F# at http://fsharp.net. See the 'F# Tutorial' project
+// for more guidance on F# programming.
+
```
Comment (dsyme) [inline src/Suave.Xsp/Script.fsx]: Should this file be removed?

File: src/Suave/Http.fs [hunk 4]
```diff
@@ -220,7 +220,7 @@ module Http =
      
 
     let not_modified : WebPart =
-      fun ctx -> { ctx with response = {status = HTTP_304; headers = []; content = Bytes [||] }} |> succeed
+      fun ctx -> { ctx with response = {status = HTTP_304; headers = []; content = Bytes [||]; writePreamble = true }} |> succeed
```
Comment (dsyme) [inline src/Suave/Http.fs]: What's this change? thanks

File: src/Suave.Xsp/README.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (dsyme) [inline src/Suave.Xsp/README.txt]: This README doesn't say much, some more info here would be great, thanks

**Comment by dsyme** [comment]
OK, so this lets you host ASP.NET applications using Suave? Is the main scenario on Mono/Linux?

I do wonder if this should go in a new package and project e.g. https://github.com/SuaveIO/Suave.Xsp.  Or under some kind of "contrib" directory.

**Comment by ademar** [comment]
> OK, so this lets you host ASP.NET applications using Suave? Is the main scenario on Mono/Linux?

Exactly.

File: src/Suave/Http.fs (hunk 4 above)
Comment (ademar) [inline src/Suave/Http.fs]: Sometimes we do not want Suave to send the HTTP preamble. Like in the Xsp module, the module is going to write directly to the socket and we do not want the default Suave workflow to execute. It is a little bit of a hack.

I think the need also arised when @haf was working on Server-Sent events.

**Comment by ademar** [comment]
I am shooting for a Linux web hosting platform somewhat ala Heroku. 

That is why I started by implementing the Proxy module, I wanted to combine Application Request Routing (ARR) and LXC containers.

**Comment by haf** [comment]
Nice, soon there will be proxies written with Suave :dancer:

File: src/Suave/Http.fs (hunk 4 above)
Comment (haf) [inline src/Suave/Http.fs]: Can we encode it as a field of SocketTask? Are there any other than SocketTask that need to avoid writing preambles?

**Comment by railcar88** [comment]
Now I get this feature.

File: src/Suave/Http.fs (hunk 4 above)
Comment (ademar) [inline src/Suave/Http.fs]: Hum .. yeah I think you are correct .. let me explore that.

Comment (ademar) [inline src/Suave/Http.fs]: Actually `SocketTask` is not good, `startWebServer` takes a `WebPart`

**Pull request merged**